odoo/addons/base/data/res_users_demo.xml

```diff
@@ -16,18 +16,22 @@
             <field name="phone">(441)-695-2334</field>
         </record>
 
-        <record id="main_partner" model="res.partner">
-            <field name="name">YourCompany</field>
-            <field name="company_name">YourCompany</field>
-            <field name="street">250 Executive Park Blvd, Suite 3400</field>
-            <field name="city">San Francisco</field>
-            <field name="zip">94134</field>
-            <field name='country_id' ref='base.us'/>
-            <field name='state_id' ref='state_us_5'/>
-            <field name="phone">+1 (650) 555-0111 </field>
-            <field name="email">info@yourcompany.example.com</field>
-            <field name="website">www.example.com</field>
-        </record>
+        <!-- Only update if we don't have information coming from the database manager -->
+        <function model="res.partner" name="write">
+            <value eval="[ref('base.main_partner')]"/>
+            <value eval="{
+                'name': 'YourCompany',
+                'street': '250 Executive Park Blvd, Suite 3400',
+                'city': 'San Francisco',
+                'zip': '94134',
+                'country_id': ref('base.us'),
+                'state_id': ref('base.state_us_5'),
+                'phone': '+1 (650) 555-0111',
+                'website': 'www.example.com',
+                'email': 'info@yourcompany.com',
+                'vat': 'US12345671',
+            } if obj(ref('base.main_partner')).name == 'My Company' else {}" model="res.partner"/>
```

Comment:
I'm OK with a fix like this in `17.0`. 👍🏻 

I don't think we currently have a more proper way to see whether the main company is "Untouched after install" than checking the name, at the moment. 🤷🏻 

Maybe in `master`, `loading.py` could ask each module how to customize the data before importing. (just like we do with Chart Templates after all). What do you think?

Reply:
Yeah, that's never going to happen I think :smile: 
It would lose the strength of the modularity through data modules etc, static code checking is less easy to do, and it probably has a lot of other impacts